From 1fba06684062a298b22801c71a1ae091d313cc11 Mon Sep 17 00:00:00 2001
From: K900 <me@0upti.me>
Date: Mon, 24 May 2021 09:57:11 +0300
Subject: [PATCH] wireguard: fix parsing keys

Load the keys from the correct place in the message, and load them as the right length as well.
---
 pyroute2/netlink/generic/wireguard.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/pyroute2/netlink/generic/wireguard.py b/pyroute2/netlink/generic/wireguard.py
index 17b510d2..74f448e0 100644
--- a/pyroute2/netlink/generic/wireguard.py
+++ b/pyroute2/netlink/generic/wireguard.py
@@ -151,11 +151,11 @@ class wgdevice_peer(nla):
                    ('WGPEER_A_PROTOCOL_VERSION', 'uint32'))
 
         class parse_peer_key(nla):
-            fields = (('key', 's'), )
+            fields = (('key', '32s'), )
 
             def decode(self):
                 nla.decode(self)
-                self['value'] = b64encode(self['value'])
+                self['value'] = b64encode(self['key'])
 
             def encode(self):
                 self['key'] = b64decode(self['value'])
@@ -220,11 +220,11 @@ def decode(self):
                 self['addr'] = '{0}/{1}'.format(self['addr'], wgaddr)
 
     class parse_wg_key(nla):
-        fields = (('key', 's'), )
+        fields = (('key', '32s'), )
 
         def decode(self):
             nla.decode(self)
-            self['value'] = b64encode(self['value'])
+            self['value'] = b64encode(self['key'])
 
         def encode(self):
             self['key'] = b64decode(self['value'])
